modules/tinymce/src/core/main/ts/fmt/FormatUtils.ts

```diff
@@ -198,6 +198,15 @@ const areSimilarFormats = (editor: Editor, formatName: string, otherFormatName:
   });
 };
 
+const isSelectorFormat = (format: ApplyFormat): format is SelectorFormat =>
+  Obj.hasNonNullableKey(format as any, 'selector');
+
+const isBlockFormat = (format: ApplyFormat): format is BlockFormat =>
+  Obj.hasNonNullableKey(format as any, 'block');
+
+const isInlineFormat = (format: ApplyFormat): format is InlineFormat =>
```

Comment:
This came up in a separate discussion in `ApplyFormat.ts` but seeing as that was resolved I'm just copying the comment over to here:
Is it a problem that `isInlineFormat` could return true for _some_ `SelectorFormat`s

Reply:
Hmm, point. I'll update the check, as I think that should be okay. I'm not really confident on how mixed formats work, which is what it's classified as when a selector has an inline attribute.

Reply:
I tried this and it does cause things to fail, so we can't do that. For now, I think I'll just revert the `inline?: string` on `SelectorFormat` and do a typecast in `ApplyFormat`. We can then try and solve the actual issues at a later stage (probably when we're fixing all the other formatter internal types issues).

Reply:
I've pushed those changes up, so see what you think. We're definitely going to have to look at formatter one day though, as we're likely missing something which is why the types don't play well.

Reply:
Looks good to me